src/libsyntax/parse/parser.rs

```diff
@@ -4273,7 +4274,9 @@ impl<'a> Parser<'a> {
                 where_clause: WhereClause {
                     id: ast::DUMMY_NODE_ID,
                     predicates: Vec::new(),
-                }
+                },
+                // We lessen by BytePos(1) because self.span is already the element following ">".
```

Comment:
nit: I'm not 100% sure I understand what this comment means.  Are you saying the span includes the ">" and we don't want it?